phc/services/workflows.py

```diff
@@ -0,0 +1,155 @@
+
+        Returns
+        -------
+        ApiResponse
+            The workflow run response
+
+        Examples
+        --------
+        >>> from phc.services import Workflows
+        >>> workflows = Workflows(session)
+        >>> workflows.run(project_id="d2876f48-724f-4987-9cf0-92c7ef99a9fa",
+              name="Ashion ingest subj: 2405",
+              tool="lifeomic/ashion-ingest-workflow",
+              workflow_inputs="{'reference': 'GRCh37','tarFile': {'class': 'File','fileId': '28235c74-9731-4496-bb3c-41c361f106f3'}, 'source': 'incoming/ashion_C043_9999_009990_T1_K1ID2_ps20190814000000.tar.gz'}")
+        """
+        create_request = {
+            "datasetId": project_id,
+            "name": name,
+            "workflowSourceFileId": tool,
+        }
+
+        if workflow_inputs:
+            create_request["workflowInputs"] = workflow_inputs
+        elif workflow_inputs_file_id:
+            create_request["workflowInputsFileId"] = workflow_inputs_file_id
+        else:
+            raise ValueError(
+                "Must provide a value for the workflow_inputs or workflow_inputs_file_id"
+            )
+
+        if output_project_folder:
+            create_request["outputProjectFolder"] = output_project_folder
+
+        res = self._api_call(
+            "/v1/workflows/ga4gh/wes/runs",
+            json=create_request,
+            http_verb="POST",
+        )
+        return res
+
+    def get(self, project_id: str, workflow_id: str) -> ApiResponse:
+        """Get workflow metadata by id
+
+        Parameters
+        ----------
+        project_id: str
+            The project ID
+        workflow_id : str
+            The workflow ID.
+
+        Returns
+        -------
+        phc.ApiResponse
+            The get workflow response
+        """
+        return self._api_call(
+            f"/v1/workflows/ga4gh/wes/runs/{project_id}:{workflow_id}",
+            http_verb="GET",
+        )
+
+    def get_list(self, project_id: str) -> ApiResponse:
+        """Fetch a list of workflows run in the specified project
+
+        Parameters
+        ----------
+        project_id: str
+            The project ID
+
+        Returns
+        -------
+        phc.ApiResponse
+            The list workflow run response
+        """
+        return self._api_call(
+            f"/v1/workflows/ga4gh/wes/runs?datasetId={project_id}",
+            http_verb="GET",
+        )
+
+    def describe(self, project_id: str, tool: str) -> ApiResponse:
+        """Fetch a list of workflows run in the specified project
+
+        Parameters
+        ----------
+        project_id: str
+            The project ID
+        tool: str
+            The tool id or organization/name of the tool to run
+
+        Returns
+        -------
+        phc.ApiResponse
+            The description of the inputs for the given tool
+        """
+        describe_request = {
+            "datasetId": project_id,
+            "workflowSourceFileId": tool,
```

Comment:
So, this parameter accepts an id or a name? Is that true in general for the API?

Reply:
Yes, through the api this is true, once I have a couple of more PR's merged, so this functionality is new.